### PR TITLE
Pull request for issue #551 to fix parsing of input fields > 512kb with JBoss

### PR DESCRIPTION
**Pull request by jodue**
Hi James,

as discussed in #551 the parsing of XML fields > 512kb fails in JBoss because the instanceof check for Woodstox failes due to some JBoss magic. I implemented and tested your proposed fix to also check the version property, which solves this issue!

### CODE REVIEW

File: hapi-fhir-base/src/main/java/ca/uhn/fhir/util/XmlUtil.java [hunk 1]
```diff
@@ -1615,7 +1615,20 @@ private static XMLInputFactory getOrCreateInputFactory() throws FactoryConfigura
...
-				if (inputFactory instanceof com.ctc.wstx.stax.WstxInputFactory) {
+				boolean isWoodstox = inputFactory instanceof com.ctc.wstx.stax.WstxInputFactory;
+				if ( !isWoodstox )
+				{
+					// Check if implementation is woodstox by property since instanceof check does not work if running in JBoss
+					try
+					{
+						isWoodstox = inputFactory.getProperty( "org.codehaus.stax2.implVersion" ) != null;
+					}
+					catch ( IllegalArgumentException e )
```
Comment (jamesagnew) [inline hapi-fhir-base/src/main/java/ca/uhn/fhir/util/XmlUtil.java]: To be extra safe, would it make sense to just catch `Exception` here?

Comment (jodue) [inline hapi-fhir-base/src/main/java/ca/uhn/fhir/util/XmlUtil.java]: According to the javadoc of XMLInputFactory [1] only an IllegalArgumentException should be thrown but if we want to be on the safe side we probably could just catch Exception.

[1] https://docs.oracle.com/javase/7/docs/api/javax/xml/stream/XMLInputFactory.html#getProperty(java.lang.String)

Comment (jamesagnew) [inline hapi-fhir-base/src/main/java/ca/uhn/fhir/util/XmlUtil.java]: Cool! Thanks for fixing that, this looks good to me. Will merge a bit later.

Comment (jodue) [inline hapi-fhir-base/src/main/java/ca/uhn/fhir/util/XmlUtil.java]: Perfect, thank you!

**Comment by jamesagnew** [comment]
Thanks again for the PR! Merging it now.

**Pull request merged**